### ISSUE TITLE
Java, UPA: IndexOutOfBoundsException thrown when applying RMTES to cache

### ISSUE THREAD

**Issue by JamesGibbon** [Refinitiv/Real-Time-SDK#96]
Certain RMTES Strings cause the RMTESApplyToCache method in RmtesDecoderImpl.java to throw an IndexOutOfBoundsException

For example the following code will cause the exception to be thrown
```java
Buffer inBuffer = CodecFactory.createBuffer();
inBuffer.data(ByteBuffer.wrap(new byte[]{27,37}));
RmtesCacheBuffer rmtesCacheBuffer = CodecFactory.createRmtesCacheBuffer(100);
RmtesDecoder rmtesDecoder = CodecFactory.createRmtesDecoder();
rmtesDecoder.RMTESApplyToCache(inBuffer, rmtesCacheBuffer);
```

The cause is this line where inBufPos is incremented without checking if it goes past the end of the buffer, and the next line which tries to retrieve the byte at that position. 

https://github.com/Refinitiv/Elektron-SDK/blob/5b873a6e6c80f983b1d54544342938d3a3f71b79/Java/Eta/Core/src/main/java/com/thomsonreuters/upa/codec/RmtesDecoderImpl.java#L942

**Comment by MitchellKato** [comment]
This case appears to be a malformed RMTES input string.  Can you please provide the context in which you're getting this string?  

Is this causing issues with production code?

The code should be erroring out correctly, not crashing.  

Thank you for reporting this, we have raised a bug in our backlog.

**Comment by JamesGibbon** [comment]
Hi Mitchell, thanks for the quick response. 

I work in Elektron Tick History. My example was contrived but I've seen a few examples of RMTES strings ending with those 2 bytes (27,37) in our historical tick data. For example this value for FID 1376 (ROW99_18):
`%0약세의 장세가 연장되는 모습으로 전환이 되었고 마지막 반전은 외환 %`

For now we can catch the exception

**Comment by vlevendel** [comment]
Fix is included with tag, Real-Time-SDK-2.0.1.L1.
